### PR TITLE
no more vars in Web/API (2)

### PR DESCRIPTION
**Pull request by lionralfs**
#### Summary

Replaces `var` in Web/API with `const` where applicable and `let` otherwise.
Mostly automated via my ESLint setup, using the rules `no-var` and `prefer-const`

#### Motivation

* See #16662
* See https://github.com/mdn/content/discussions/10389
* See https://github.com/orgs/mdn/discussions/143

#### Metadata

- [ ] Adds a new document
- [ ] Rewrites (or significantly expands) a document
- [x] Fixes a typo, bug, or other error

### CODE REVIEW

File: files/en-us/web/api/cssstylesheet/insertrule/index.md [hunk 1]
```diff
@@ -135,16 +135,16 @@ to the default native `insertRule()`.
...
   if (originalInsertRule.length === 2){ // 2 mandatory arguments: (selector, rules)
     Sheet_proto.insertRule = function(selectorAndRule){
       // First, separate the selector from the rule
-      a: for (var i=0, Len=selectorAndRule.length, isEscaped=0, newCharCode=0; i !== Len; ++i) {
+      a: for (let i=0, Len=selectorAndRule.length, isEscaped=0, newCharCode=0; i !== Len; ++i) {
         newCharCode = selectorAndRule.charCodeAt(i);
         if (!isEscaped && (newCharCode === 123)) { // 123 = "{".charCodeAt(0)
           // Secondly, find the last closing bracket
-          var openBracketPos = i, closeBracketPos = -1;
+          let openBracketPos = i, closeBracketPos = -1;
```
Comment (lionralfs) [inline files/en-us/web/api/cssstylesheet/insertrule/index.md:147]: I should probably revert these, as they are in a polyfill section mentioning Internet Explorer 5–8.

Comment (lionralfs) [reply files/en-us/web/api/cssstylesheet/insertrule/index.md:147]: Might also consider something similar to `example-bad`. Relevant discussion here: https://github.com/eslint/eslint-plugin-markdown/issues/208

File: files/en-us/web/api/document/lastmodified/index.md [hunk 2]
```diff
@@ -65,7 +65,7 @@ if (Date.parse(document.lastModified) > parseFloat(document.cookie.replace(/(?:(
 …the same example, but skipping the first visit:
 
 ```js
-var
+const
```
[the remark this answers is not in the document]
Comment (lionralfs) [reply files/en-us/web/api/document/lastmodified/index.md:68]: Sure. I'll also rename them

File: files/en-us/web/api/document/registerelement/index.md [hunk 3]
```diff
@@ -47,7 +47,7 @@ None ({{jsxref("undefined")}}).
 Here is a very simple example:
 
 ```js
-var Mytag = document.registerElement('my-tag');
+const Mytag = document.registerElement('my-tag');
```
[the remark this answers is not in the document]
Comment (lionralfs) [reply files/en-us/web/api/document/registerelement/index.md:50]: I think this should stay the way it is, the following section reads:

> Now the new tag is registered in the browser. The `Mytag` variable holds a
> constructor that you can use to create a `my-tag` element in the document as
> follows:
> 
> ```js
> document.body.appendChild(new Mytag());
> ```

File: files/en-us/web/api/baseaudiocontext/currenttime/index.md [hunk 4]
```diff
@@ -25,8 +25,8 @@ A floating point number.
 ## Examples
 
 ```js
-var AudioContext = window.AudioContext || window.webkitAudioContext;
-var audioCtx = new AudioContext();
+const AudioContext = window.AudioContext || window.webkitAudioContext;
```
Comment (teoli2003) [inline files/en-us/web/api/baseaudiocontext/currenttime/index.md:28]:
```suggestion
```

File: files/en-us/web/api/baseaudiocontext/destination/index.md [hunk 5]
```diff
@@ -29,12 +29,12 @@ An {{ domxref("AudioDestinationNode") }}.
 > Demos on the [MDN GitHub repo](https://github.com/mdn/), like [voice-change-o-matic](https://github.com/mdn/voice-change-o-matic).
 
 ```js
-var AudioContext = window.AudioContext || window.webkitAudioContext;
-var audioCtx = new AudioContext();
+const AudioContext = window.AudioContext || window.webkitAudioContext;
```
Comment (teoli2003) [inline files/en-us/web/api/baseaudiocontext/destination/index.md:32]:
```suggestion
```

File: files/en-us/web/api/baseaudiocontext/listener/index.md [hunk 6]
```diff
@@ -28,13 +28,13 @@ An {{ domxref("AudioListener") }} object.
 > **Note:** for a full Web Audio spatialization example, see our [panner-node](https://github.com/mdn/webaudio-examples/tree/master/panner-node) demo.
 
 ```js
-var AudioContext = window.AudioContext || window.webkitAudioContext;
-var audioCtx = new AudioContext();
+const AudioContext = window.AudioContext || window.webkitAudioContext;
```
Comment (teoli2003) [inline files/en-us/web/api/baseaudiocontext/listener/index.md:31]:
```suggestion
```

File: files/en-us/web/api/baseaudiocontext/samplerate/index.md [hunk 7]
```diff
@@ -31,8 +31,8 @@ second.
 > `audioCtx.sampleRate` into your browser console.
 
 ```js
-var AudioContext = window.AudioContext || window.webkitAudioContext;
-var audioCtx = new AudioContext();
+const AudioContext = window.AudioContext || window.webkitAudioContext;
```
Comment (teoli2003) [inline files/en-us/web/api/baseaudiocontext/samplerate/index.md:34]:
```suggestion
```

File: files/en-us/web/api/biquadfilternode/detune/index.md [hunk 8]
```diff
@@ -26,14 +26,14 @@ An [a-rate](/en-US/docs/Web/API/AudioParam#a-rate) {{domxref("AudioParam")}}.
 The following example shows basic usage of an AudioContext to create a Biquad filter node. For a complete working example, check out our [voice-change-o-matic](https://mdn.github.io/voice-change-o-matic/) demo (look at the [source code](https://github.com/mdn/voice-change-o-matic) too).
 
 ```js
-var audioCtx = new (window.AudioContext || window.webkitAudioContext)();
+const audioCtx = new (window.AudioContext || window.webkitAudioContext)();
```
Comment (teoli2003) [inline files/en-us/web/api/biquadfilternode/detune/index.md:29]:
```suggestion
const audioCtx = new AudioContext();
```

File: files/en-us/web/api/biquadfilternode/frequency/index.md [hunk 9]
```diff
@@ -28,14 +28,14 @@ An {{domxref("AudioParam")}}.
 The following example shows basic usage of an AudioContext to create a Biquad filter node. For a complete working example, check out our [voice-change-o-matic](https://mdn.github.io/voice-change-o-matic/) demo (look at the [source code](https://github.com/mdn/voice-change-o-matic) too).
 
 ```js
-var audioCtx = new (window.AudioContext || window.webkitAudioContext)();
+const audioCtx = new (window.AudioContext || window.webkitAudioContext)();
```
Comment (teoli2003) [inline files/en-us/web/api/biquadfilternode/frequency/index.md:31]:
```suggestion
const audioCtx = new AudioContext();
```

File: files/en-us/web/api/biquadfilternode/q/index.md [hunk 10]
```diff
@@ -28,14 +28,14 @@ An {{domxref("AudioParam")}}.
 The following example shows basic usage of an AudioContext to create a Biquad filter node. For a complete working example, check out our [voice-change-o-matic](https://mdn.github.io/voice-change-o-matic/) demo (look at the [source code](https://github.com/mdn/voice-change-o-matic) too).
 
 ```js
-var audioCtx = new (window.AudioContext || window.webkitAudioContext)();
+const audioCtx = new (window.AudioContext || window.webkitAudioContext)();
```
Comment (teoli2003) [inline files/en-us/web/api/biquadfilternode/q/index.md:31]:
```suggestion
const audioCtx = new AudioContext();
```

File: files/en-us/web/api/biquadfilternode/type/index.md [hunk 11]
```diff
@@ -168,14 +168,14 @@ A string (enum) representing a [BiquadFilterType](https://webaudio.github.io/web
 The following example shows basic usage of an AudioContext to create a Biquad filter node. For a complete working example, check out our [voice-change-o-matic](https://mdn.github.io/voice-change-o-matic/) demo (look at the [source code](https://github.com/mdn/voice-change-o-matic) too).
 
 ```js
-var audioCtx = new (window.AudioContext || window.webkitAudioContext)();
+const audioCtx = new (window.AudioContext || window.webkitAudioContext)();
```
Comment (teoli2003) [inline files/en-us/web/api/biquadfilternode/type/index.md:171]:
```suggestion
const audioCtx = new AudioContext();
```

File: files/en-us/web/api/biquadfilternode/gain/index.md [hunk 12]
```diff
@@ -30,14 +30,14 @@ An {{domxref("AudioParam")}}.
 The following example shows basic usage of an AudioContext to create a Biquad filter node. For a complete working example, check out our [voice-change-o-matic](https://mdn.github.io/voice-change-o-matic/) demo (look at the [source code](https://github.com/mdn/voice-change-o-matic) too).
 
 ```js
-var audioCtx = new (window.AudioContext || window.webkitAudioContext)();
+const audioCtx = new (window.AudioContext || window.webkitAudioContext)();
```
Comment (teoli2003) [inline files/en-us/web/api/biquadfilternode/gain/index.md:33]:
```suggestion
const audioCtx = new AudioContext();
```

File: files/en-us/web/api/bluetooth/getavailability/index.md [hunk 13]
```diff
@@ -21,7 +21,7 @@ The **`getAvailability()`** method of the {{DOMxRef("Bluetooth")}} interface ret
 ## Syntax
 
 ```js
-var readerPromise = Bluetooth.getAvailability();
+const readerPromise = Bluetooth.getAvailability();
```
Comment (teoli2003) [inline files/en-us/web/api/bluetooth/getavailability/index.md:24]:
```suggestion
getAvailability()
```

File: files/en-us/web/api/bluetooth/getdevices/index.md [hunk 14]
```diff
@@ -23,7 +23,7 @@ permission prompts.
 ## Syntax
 
 ```js
-var readerPromise = Bluetooth.getDevices();
+const readerPromise = Bluetooth.getDevices();
```
Comment (teoli2003) [inline files/en-us/web/api/bluetooth/getdevices/index.md:26]:
```suggestion
getDevices()
```

File: files/en-us/web/api/bluetoothremotegattdescriptor/characteristic/index.md [hunk 15]
```diff
@@ -23,7 +23,7 @@ descriptor belongs to.
 ## Syntax
 
 ```js
```
Comment (teoli2003) [inline files/en-us/web/api/bluetoothremotegattdescriptor/characteristic/index.md:25]:
```suggestion
```

File: files/en-us/web/api/bluetoothremotegattdescriptor/characteristic/index.md [hunk 16]
```diff
@@ -23,7 +23,7 @@ descriptor belongs to.
...
-var characteristic = BluetoothRemoteGATTCharacteristic.characteristic
+const characteristic = BluetoothRemoteGATTCharacteristic.characteristic
 ```
 
 ### Returns
```
Comment (teoli2003) [inline files/en-us/web/api/bluetoothremotegattdescriptor/characteristic/index.md:29]:
```suggestion
## Value
```

File: files/en-us/web/api/canvascapturemediastreamtrack/canvas/index.md [hunk 17]
```diff
@@ -29,16 +29,16 @@ frames being captured.
 
 ```js
 // Find the canvas element to capture
-var canvasElt = document.getElementsByTagName("canvas")[0];
+const canvasElt = document.getElementsByTagName("canvas")[0];
```
Comment (teoli2003) [inline files/en-us/web/api/canvascapturemediastreamtrack/canvas/index.md:32]:
```suggestion
const canvasElt = document.getQuerySelector("canvas");
```

File: files/en-us/web/api/canvascapturemediastreamtrack/requestframe/index.md [hunk 18]
```diff
@@ -51,10 +51,10 @@ mentioned in the spec, this is a likely candidate).
 
 ```js
 // Find the canvas element to capture
-var canvasElt = document.getElementsByTagName("canvas")[0];
+const canvasElt = document.getElementsByTagName("canvas")[0];
```
Comment (teoli2003) [inline files/en-us/web/api/canvascapturemediastreamtrack/requestframe/index.md:54]:
```suggestion
const canvasElt = document.getQuerySelector("canvas");
```

File: files/en-us/web/api/clipboarditem/gettype/index.md [hunk 19]
```diff
@@ -20,7 +20,7 @@ The **`getType()`** method of the {{domxref("ClipboardItem")}} interface returns
 ## Syntax
 
 ```js
-var blob = clipboardItem.getType(type);
+const blob = clipboardItem.getType(type);
```
Comment (teoli2003) [inline files/en-us/web/api/clipboarditem/gettype/index.md:23]:
```suggestion
getType(type)
```

File: files/en-us/web/api/contactsmanager/getproperties/index.md [hunk 20]
```diff
@@ -21,7 +21,7 @@ properties are available.
 ## Syntax
 
 ```js
-var ContactProperties = ContactsManager.getProperties();
+const contactProperties = ContactsManager.getProperties();
```
Comment (teoli2003) [inline files/en-us/web/api/contactsmanager/getproperties/index.md:24]:
```suggestion
getProperties()
```

File: files/en-us/web/api/cssstyledeclaration/msgetpropertyenabled/index.md [hunk 21]
```diff
@@ -18,7 +18,7 @@ This proprietary method is specific to Internet Explorer browser.
 ### Syntax
 
 ```js
-var retVal = style.msGetPropertyEnabled(name);
+const retVal = style.msGetPropertyEnabled(name);
```
Comment (teoli2003) [inline files/en-us/web/api/cssstyledeclaration/msgetpropertyenabled/index.md:21]:
```suggestion
msGetPropertyEnabled(name)
```

File: files/en-us/web/api/cssstyledeclaration/msputpropertyenabled/index.md [hunk 22]
```diff
@@ -18,7 +18,7 @@ This proprietary method is specific to Internet Explorer and Microsoft Edge.
 ### Syntax
 
 ```js
-var retval = style.msPutPropertyEnabled(propertyName, true);
+const retval = style.msPutPropertyEnabled(propertyName, true);
```
Comment (teoli2003) [inline files/en-us/web/api/cssstyledeclaration/msputpropertyenabled/index.md:21]:
```suggestion
msPutPropertyEnabled(name, boolean)
```

File: files/en-us/web/api/datatransferitem/getasfilesystemhandle/index.md [hunk 23]
```diff
@@ -20,7 +20,7 @@ dragged item is a directory.
 ## Syntax
 
 ```js
-var handle = DataTransferItem.getAsFileSystemHandle();
+const handle = DataTransferItem.getAsFileSystemHandle();
```
Comment (teoli2003) [inline files/en-us/web/api/datatransferitem/getasfilesystemhandle/index.md:23]:
```suggestion
getAsFileSystemHandle()
```

**Review by teoli2003 — CHANGES_REQUESTED** [review]

File: files/en-us/web/api/canvascapturemediastreamtrack/canvas/index.md (hunk 17 above)
Comment (lionralfs) [reply files/en-us/web/api/canvascapturemediastreamtrack/canvas/index.md:32]:
```suggestion
const canvasElt = document.querySelector("canvas");
```
?

File: files/en-us/web/api/canvascapturemediastreamtrack/requestframe/index.md (hunk 18 above)
Comment (lionralfs) [reply files/en-us/web/api/canvascapturemediastreamtrack/requestframe/index.md:54]:
```suggestion
const canvasElt = document.querySelector("canvas");
```

**Review by teoli2003 — APPROVED** [review]
👍

**Pull request merged**